### DISCUSSION TITLE
Namelists in esm-tools

### DISCUSSION THREAD

**Discussion by pgierz** [esm-tools/esm_tools#669] [Polls]
This has been a question we have often thought about, and has been named often enough in meetings. Should we, as the `esm-tools` maintainers, give you namelists for your model?

**Comment by seb-wahl** [comment]
For NEMO, we actually already use option 2 to be as independent of ESM-Tools as possible.

**Reply by pgierz** [reply]
I also would agree with this strategy. As far as I know, this would eliminate the need to keep FESOM namelists around (maybe also ECHAM? I am not sure at GEOMAR, but at least our ECHAM does not give namelist examples)

**Comment by JanStreffing** [comment]
I think this sounds like a reasonable idea. However it is not what we do for AWI-CM3 at the moment. So some work would be needed on the source repo side.

**Comment by christian-stepanek** [comment]
A very important point is that there should be easy-to-set switches via the experiment-YAMLs to change namelist settings (sometimes even during a simulation, for example to switch the output mode from "spinup" to "production").

**Reply by mandresm** [reply]
Cannot you define those switches already? or is this a problem with the ECHAM streams?

**Reply by christian-stepanek** [reply]
I studied the documentation and there is 16.10 Using your own namelist". Yet, is this the most sensible way to switch between different namelists, for example in order to switch from output for a spinup to output for a production or output for a CMORization.

If this is already possible/documented (e.g. via the "scenario" settings), then my apologies, but I did not find it.

**Reply by mandresm** [reply]
You are right, the scenario settings are not described in the ESM-Tools documentation. I think they are documented in the AWIESM documentation @pgierz put together. Something I'd like to add here is that `scenarios` is not a ESM-Tools featured variable (i.e. does not trigger any action natively from ESM-Tools, ESM-Tools doesn't "know" what scenario means). Instead, it is used in several models to activate configurations in `choose_` blocks (i.e. switches), so it's functionality it's exclusively defined in the yaml files.

As you know, on May we met with you, the AWIESM people, for organizing a cleanup and making decisions about how to operate AWIESM-2. Then it was decided that the namelists will be provided by ESM-Tools for different scenarios and depending on which scenario, one namelist will be used or another. This is already in release.

Of course we can always change whether the namelists are provided by ESM-Tools, or that they sit somewhere else (come together with the model), that's ultimately your choice. Independently of where the namelists are stored, it is always possible to add more scenarios and namelists, or even other types of switches, by just changing the `echam.yaml`, `jsbach.yaml` or `awiesm.yaml`.

In summary, the difficulty I see is not on technically setting up the switches (this takes very little time), putting the namelists here or there; but instead on obtaining the specific information from the different modeling groups on how to operate their systems (which scenarios? which switches you need? where do you want to store your namelists...). If you want some new switches for easily changing the namelists just let us know via an issue (or you can set them up yourself and pull-request), we will be happy to help configuring those :)

And a final remark, as you know, where we actually have more "technical" difficulties is on modifying ECHAM namelists for the streams, as our namelist parser (3rd party module) does not allow for namelist sections named the same. This could be solved in some point by us (@pgierz already put together some thoughts and try some approaches on that), but for the time being it seems that the "switch" approach is the most reasonable one.

**Reply by mandresm** [reply]
By the way, I have an idea for switching between different outputs (even during the same simulation, from spinup to production). The idea would be basically to split the stream part from the rest of the namelist, and let ESM-Tools put them together back. In this way we can have a bunch of output cases that can be plugged in to the final namelist with just a switch. I'll talk to @pgierz about it on our next Monday meeting, and we will come back to you.

**Reply by christian-stepanek** [reply]
Great! Thanks!

**Comment by mandresm** [comment]
I think this should be a decision made by the model developers, for each particular model
